File: Data Structures & Algorithms/jump-game/submission-0.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        end = 0
        for i in range(len(nums)):
            if i > end:
                # print('break')
                break
            if nums[i] > 0:
                end = max(end, i + nums[i])
        # print(end)
        return end >= len(nums) - 1

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int n = nums.length;
        int[] leftProducts = new int[n];
        int[] rightProducts = new int[n];
        int cur = 1;
        for (int i = 0; i < n; i++) {
            leftProducts[i] = cur;
            cur *= nums[i];
        }
        cur = 1;
        for (int i = n - 1; i >= 0; i--) {
            rightProducts[i] = cur;
            cur *= nums[i];
        }
        int[] results = new int[n];
        for (int i = 0; i < n; i++) {
            results[i] = leftProducts[i] * rightProducts[i];
        }
        return results;
    }
}  
